decoder/vaapidecoder_base.h

```diff
@@ -52,12 +52,15 @@ struct VideoDecoderConfig {
 
     VideoDecoderConfig()
     {
-        resetConfig();
+        //resetConfig();
+       memset(this, 0, sizeof(*this));
+        profile = VAProfileNone;
     }
     void resetConfig()
     {
-        memset(this, 0, sizeof(*this));
-        profile = VAProfileNone;
+       // memset(this, 0, sizeof(*this));
+       // profile = VAProfileNone;
+       VideoDecoderConfig();
```

Comment:
call construct in other fucntion many not allowed in some gcc version. could we just use
memset(this, 0, offset(VideoDecoderConfig, profile)); profile =VAProfileNone

Reply:
it will occurs the filed error,
vaapidecoder_base.h:60:61: error: ‘void* memset(void*, int, size_t)’ clearing an object of non-trivial type ‘struct YamiMediaCodec::VideoDecoderConfig’; use assignment or value-initialization instead [-Werror=class-memaccess]
        memset(this, 0, offsetof(VideoDecoderConfig, profile));

How about this
void resetConfig()
    {
       width=0;
       height=0;
       fourcc=0;
       surfaceNumber=0;
       profile = VAProfileNone;
    }